library/ziti.c

```diff
@@ -638,19 +656,43 @@ void ziti_force_session_refresh(ziti_context ztx) {
 
 void ziti_re_auth_with_cb(ziti_context ztx, void(*cb)(ziti_session *, const ziti_error *, void *), void *ctx) {
     ZTX_LOG(WARN, "starting to re-auth with ctlr[%s]", ztx->opts->controller);
+
+    //to force a re-auth, set api_session_sate = ZitiApiSessionStateUnauthenticated
+    if (ztx->api_session_state == ZitiApiSessionStatePartiallyAuthenticated || ztx->api_session_state == ZitiApiSessionStateAuthStarted) {
```

Comment:
do we need to check if api_session has expired and ztx need to restart auth from very begining?

Reply:
Yea I've enhanced this logic to check for that condition.